docs/implementing-cards.md

```diff
@@ -1,388 +1,570 @@
 });
 ```
 
-#### Conditional effects
+If not provided, `matchTarget` will default to targeting only the card that owns the constant ability.
+
+#### Conditional ongoing effects
 
-Some effects have a 'when', 'while' or 'if' clause within their text. These cards can be implemented by passing a `condition` function into the persistent effect declaration. The effect will only be applied when the function returns `true`. If the function returns `false` later on, the effect will be automatically unapplied from the cards it matched.
+Some ongoing effects have a 'when', 'while' or 'if' clause within their text. These cards can be implemented by passing a `condition` function into the constant ability declaration. The ongoing effect will only be applied when the function returns `true`. If the function returns `false` later on, the ongoing effect will be automatically unapplied from the cards it matched.
 
 ```javascript
-// During a conflict in which this character is participating, each other participating Lion 
-// character you control gets +1M.
-this.persistentEffect({
-    condition: () => this.isParticipating(),
-    match: card => card.getType() === 'character' && card.isParticipating() && 
-                   card.isFaction('lion') && card !== this,
-    effect: AbilityDsl.effects.modifyMilitarySkill(1)
+// While this unit is exhausted, it gains +1/+1
+this.constantAbility({
+    condition: () => this.exhausted,
+    ongoingEffect: AbilityHelper.ongoingEffects.modifyStats({ power: 1, hp: 1 })
 });
 ```
 
-#### Targeting opponent or all matching cards
-
-By default, an effect will only be applied to cards controlled by the current player. The `targetController` property can be modified to specify which players' cards should be targeted.
+#### Filtering by card type, owner, location
+Note also that, similar to target resolvers described below, there are shorthand filters for the card properties location, owner, and card type. See section [Target filtering](#target-filtering) below for more details.
 
-To target only opponent cards, set `targetController` to `'opponent'`:
+All of these filters are available for filtering target cards (e.g., `targetLocationFilter`), but for checking the properties of the source card (the card that owns the ability) only `sourceLocationFilter` is available:
 
 ```javascript
-// While  attacking, each defending character gets -1M.
-this.persistentEffect({
-    condition: () => this.isAttacking(),
-    match: card => card.isDefending(),
-    targetController: 'opponent',
-    effect: AbilityDsl.effects.modifyMilitarySkill(-1)
+// While this card is in the ground arena, all of the opponent's units in the space arena get -1/-1
+this.constantAbility({
+    sourceLocationFilter: Location.GroundArena,
+    targetLocationFilter: Location.SpaceArena,
+    targetCardType: WildcardCardType.Unit,
+    targetController: RelativePlayer.Opponent,
+    ongoingEffect: AbilityHelper.ongoingEffects.modifyStats({ power: -1, hp: -1 })
```

Comment:
modifyStats should be 1, 1 not -1, -1

Reply:
Why is that? It's just a random example, and the comment says -1/-1

Reply:
I must have misread the comment as 1/1. My b